### DISCUSSION TITLE
How to handle api shorturl request with & in the url

### DISCUSSION THREAD

**Discussion by willmayes** [YOURLS/YOURLS#3010] [Q&A] [answered]
Hi - I'm posting a URL with "&" in the url query:
https://www.exampledomain.com/p/yourls-api.php?signature=xxxxxxxxxx&action=shorturl&format=simple&url=https://example.com/testing.asp?ID=123&OrderID=ABC&...

This obviously results in only the url "https://example.com/testing.asp?ID=123" being shortened. Whats the best way to handle this?

**Comment by dgw** [comment, marked as answer]
[`urlencode()`](https://www.php.net/manual/en/function.urlencode.php) or the equivalent in whatever language you're using to call the API.
